comments/comments.controller.js

```diff
@@ -0,0 +1,74 @@
+const express = require("express");
+const router = express.Router();
+const {
+  getComment,
+  addComment,
+  updateComment,
+  removeComment
+} = require("./comments.service");
+const { getUser } = require("../users/users.service");
+
+router.get("/:id", async (req, res) => {
+  try {
+    const { id } = req.params;
+    const comment = await getComment(id);
+  
+    if (!comment)
+      return res.status(404).send("The comment with the given ID was not found");
+  
+    res.send(comment);
+  } catch (error) {
+    res.status(500).send(error.message);
+  }
+});
+
+router.post("/", async (req, res) => {
+  try {
+    const { userId, datePosted, content } = req.body;
+
+    if (!userId || userId === "") return res.status(404).send("No game ID found");
+
+    const userObject = await getUser(userId);
+
+    const comment = await addComment(userObject, datePosted, content);
+
+    res.status(201).send(comment);
+  } catch (error) {
+    res.status(500).send(error.message);
+  }
+});
+
+router.put("/:id", async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { userId, datePosted, content } = req.body;
+
+    const userObject = await getUser(userId);
+
+    const comment = await updateComment(id, {
+      userObject,
+      datePosted,
+      content
+    });
+
+    if (!comment)
+      return res.status(404).send("The comment with the given ID was not found");
```

Comment:
It might not necessary be the id that is the issue - perhaps return a general message ('unable to update the comment...')